.github/workflows/codegen.yaml

```diff
@@ -0,0 +1,78 @@
+name: Generated code
+on:
+  push:
+    branches:
+    - 'master'
+  pull_request:
+    branches:
+      - 'master'
+env:
+  # Golang version to use
+  GOLANG_VERSION: 1.16.4
+  # Version of operator-sdk binary
+  SDK_VERSION: 1.11.0
+  # Checksum of operator-sdk binary
+  SDK_CHECKSUM: 'b1f6fb02c619cfcdb46edf41cbeb4d66f627fd8bba122edaeeb06718965299eb'
+
+jobs:
+  check-go-modules:
+    name: "Check for go.mod/go.sum synchronicity"
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+      - name: Setup Golang
+        uses: actions/setup-go@v1
+        with:
+          go-version: ${{ env.GOLANG_VERSION }}
+      - name: Download all Go modules
+        run: |
+          go mod download
+      - name: Check for tidyness of go.mod and go.sum
+        run: |
+          go mod tidy
+          git diff --exit-code -- .
+
+  check-sdk-codegen:
+    name: "Check for changes from make bundle"
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+      - name: Setup Golang
+        uses: actions/setup-go@v1
+        with:
+          go-version: ${{ env.GOLANG_VERSION }}
+      - name: Download and install Operator SDK
+        run: |
+          set -ue
+          set -o pipefail
+          curl -sLf --retry 3 \
+            -o /tmp/operator-sdk_linux_amd64 \
+            https://github.com/operator-framework/operator-sdk/releases/download/v1.11.0/operator-sdk_linux_amd64
+          calculated=$(sha256sum /tmp/operator-sdk_linux_amd64 | awk '{print $1}')
+          if test "${calculated}" != "${SDK_CHECKSUM}"; then
+            echo "FAILED TO VALIDATE CHECKSUM" >&2
+            echo "Download is: ${calculated}"
+            echo "Should: ${SDK_CHECKSUM}"
+            exit 1
+          fi
+          sudo install -m 0755 /tmp/operator-sdk_linux_amd64 /usr/local/bin/operator-sdk
+      - name: Run make bundle
+        run: |
+          make bundle
+      - name: Ensure there is no diff in bundle
+        run: |
+          git diff --exit-code -- .
+      - name: Run make generate
```

Comment:
The `zz_generated.deepcopy.go` created by `make generate` will be different if the developer is using golang 1.17 because of this:
_//go:build lines
The go command now understands //go:build lines and prefers them over // +build lines. The new syntax uses boolean expressions, just like Go, and should be less error-prone. As of this release, the new syntax is fully supported, and all Go files should be updated to have both forms with the same meaning. **To aid in migration, gofmt now automatically synchronizes the two forms.** For more details on the syntax and migration plan, see https://golang.org/design/draft-gobuild._
from here: https://golang.org/doc/go1.17

That's not a reason not to add these checks, but it's something to be aware of.

Reply:
I could not reproduce it. On the latest commit (which was generated with go1.16.3):

```
$ go version
go version go1.17.1 linux/amd64
$ git show HEAD -s
commit cbf7b3295c503f76386473100344a16338035635 (HEAD -> master, upstream/master)
Author: jannfis <jann@mistrust.net>
Date:   Thu Sep 9 23:35:21 2021 +0200

    chore: Add a check for generated code (#438)
    
    * chore: Add a check for generated code
$ git status
On branch master
nothing to commit, working tree clean
$ make generate
/home/vagrant/src/argocd-operator/bin/controller-gen object:headerFile="hack/boilerplate.go.txt" paths="./..."
$ git status
On branch master
nothing to commit, working tree clean
```

Reply:
Ah, it's actually `go fmt` who does this

Reply:
Weird, that's not what I saw:
```
% go version
go version go1.17 darwin/amd64
% git status
On branch master
Your branch is up to date with 'origin/master'.
nothing to commit, working tree clean
% make generate
/Users/jpitman/tmp/foo/foo/argocd-operator/bin/controller-gen object:headerFile="hack/boilerplate.go.txt" paths="./..."
% git status
On branch master
Your branch is up to date with 'origin/master'.
Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
	modified:   api/v1alpha1/zz_generated.deepcopy.go
no changes added to commit (use "git add" and/or "git commit -a")
```